tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Wcf/AsyncMethodInvoker_InvokeEnd_Integration.cs

```diff
@@ -39,16 +41,22 @@ public class AsyncMethodInvoker_InvokeEnd_Integration
         /// <returns>A response value, in an async scenario will be T of Task of T</returns>
         internal static CallTargetReturn<TReturn> OnMethodEnd<TTarget, TReturn>(TTarget instance, TReturn returnValue, Exception exception, in CallTargetState state)
         {
-            var operationContext = WcfCommon.GetCurrentOperationContext?.Invoke();
-
-            if (operationContext != null && operationContext.TryDuckCast<IOperationContextStruct>(out var operationContextProxy))
+            if (exception is not null)
             {
-                var requestContext = operationContextProxy.RequestContext;
+                var operationContext = WcfCommon.GetCurrentOperationContext?.Invoke();
 
-                // Retrieve the scope that we saved during InvokeBegin
-                if (WcfCommon.Scopes.TryGetValue(((IDuckType)requestContext).Instance, out var scope))
+                if (operationContext != null && operationContext.TryDuckCast<IOperationContextStruct>(out var operationContextProxy))
                 {
-                    scope.DisposeWithException(exception);
+                    var requestContext = operationContextProxy.RequestContext;
+
+                    // Retrieve the scope that we saved during InvokeBegin
+                    if (((IDuckType?)requestContext)?.Instance is object requestContextInstance
+                        && WcfCommon.Scopes.TryGetValue(requestContextInstance, out var scope))
+                    {
+                        // Add the exception but do not dispose the span.
+                        // BeforeSendReplyIntegration is responsible for closing the span.
+                        scope.Span?.SetException(exception);
```

Comment:
Note: The new AfterReceiveRequestIntegration and BeforeSendReplyIntegration integrations start/stop the span. This will only decorate the existing span with exception information.